controllers/etcdpeer_controller.go

```diff
@@ -162,6 +179,53 @@ func defineReplicaSet(peer etcdv1alpha1.EtcdPeer) appsv1.ReplicaSet {
 	}
 }
 
+func pvcForPeer(peer *etcdv1alpha1.EtcdPeer) *corev1.PersistentVolumeClaim {
+	return &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      peer.Name,
+			Namespace: peer.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(peer, etcdv1alpha1.GroupVersion.WithKind("EtcdPeer")),
+			},
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.ReadWriteMany,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					"storage": peer.Spec.VolumeClaimTemplate.Spec.Resources.Requests["storage"],
+				},
+			},
+			StorageClassName: pointer.StringPtr(*peer.Spec.VolumeClaimTemplate.Spec.StorageClassName),
+		},
+	}
+}
+
+func (r *EtcdPeerReconciler) getOrCreatePvc(ctx context.Context, peer *etcdv1alpha1.EtcdPeer) (pvc *corev1.PersistentVolumeClaim, created bool, err error) {
+	objectKey := client.ObjectKey{
+		Name:      peer.Name,
+		Namespace: peer.Namespace,
+	}
+	// Check for existing object
+	pvc = &corev1.PersistentVolumeClaim{}
+	err = r.Get(ctx, objectKey, pvc)
+	// Object exists
+	if err == nil {
+		return
```

Comment:
nit: i know its not necessary, but for clarity can we explicitly return some values here? i.e.
 ```
if err == nil {
    return pvc, false, nil
}
```

Reply:
Yeah, I agree. I was trying to be clever.